test/e2e/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"flag"
+	"github.com/golang/glog"
+	osconfigv1 "github.com/openshift/api/config/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	capiv1alpha1 "sigs.k8s.io/cluster-api/pkg/apis/cluster/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+)
+
+const (
+	namespace = "openshift-cluster-api"
+)
+
+func init() {
+	if err := capiv1alpha1.AddToScheme(scheme.Scheme); err != nil {
+		glog.Fatal(err)
+	}
+
+	if err := osconfigv1.AddToScheme(scheme.Scheme); err != nil {
+		glog.Fatal(err)
+	}
+}
+
+type testConfig struct {
+	client client.Client
+}
+
+func newClient() (client.Client, error) {
+	// Get a config to talk to the apiserver
+	cfg, err := config.GetConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	return client.New(cfg, client.Options{})
+
+}
+
+func main() {
+	flag.Parse()
+	if err := runSuite(); err != nil {
+		glog.Fatal(err)
+	}
+}
+
+func runSuite() error {
+
+	client, err := newClient()
+	if err != nil {
+		return err
+	}
+	testConfig := &testConfig{
+		client: client,
+	}
+	glog.Info("RUN: ExpectProviderAvailable")
+	if err := testConfig.ExpectProviderAvailable(); err != nil {
+		glog.Errorf("FAIL: ExpectProviderAvailable: %v", err)
+		return err
+	}
+	glog.Info("PASS: ExpectProviderAvailable")
+
+	glog.Info("RUN: ExpectOneClusterObject")
+	if err := testConfig.ExpectOneClusterObject(); err != nil {
+		glog.Errorf("FAIL: ExpectOneClusterObject: %v", err)
+		return err
+	}
+	glog.Info("PASS: ExpectOneClusterObject")
+
+	glog.Info("RUN: ExpectReconcileControllersDeployment")
```

Comment:
It's fine to have this here but this is actually validating machine api operator behaviour. We'll want to battle prove the machineSet/machine(actuator) controllers here e.g https://github.com/openshift/machine-api-operator/blob/master/test/e2e/operator_expectations.go#L89
https://github.com/openshift/machine-api-operator/blob/master/test/e2e/operator_expectations.go#L178
And also will want to include coverage for the validations we do against a vanilla cluster https://github.com/openshift/cluster-api-provider-aws/blob/master/test/machines/machines_test.go#L129